pivy-importer/src/main/groovy/com/linkedin/python/importer/ImporterCLI.java

```diff
@@ -57,26 +59,45 @@ public static void main(String[] args) throws Exception {
             throw new RuntimeException("Unable to continue, no repository location given on the command line (use the --repo switch)");
         }
         final File repoPath = new File(line.getOptionValue("repo"));
-        final DependencySubstitution replacements = new DependencySubstitution(buildSubstitutionMap(line), buildForceMap(line));
 
         repoPath.mkdirs();
 
         if (!repoPath.exists() || !repoPath.isDirectory()) {
             throw new RuntimeException("Unable to continue, " + repoPath.getAbsolutePath() + " does not exist, or is not a directory");
         }
 
+        importPackages(line, repoPath);
+        logger.info("Execution Finished!");
+    }
 
+    private static void importPackages(CommandLine line, File repoPath) {
+        final DependencySubstitution replacements = new DependencySubstitution(buildSubstitutionMap(line), buildForceMap(line));
         Set<String> processedDependencies = new HashSet<>();
         for (String dependency : line.getArgList()) {
-            DependencyDownloader dependencyDownloader = new DependencyDownloader(
-                    dependency, repoPath, replacements, line.hasOption("latest"), line.hasOption("pre"),
-                    line.hasOption("lenient"));
-            dependencyDownloader.getProcessedDependencies().addAll(processedDependencies);
-            dependencyDownloader.download();
-            processedDependencies.addAll(dependencyDownloader.getProcessedDependencies());
-        }
+            DependencyDownloader artifactDownloader;
+
+            if (dependency.split(":").length == 2) {
+                artifactDownloader = new SdistDownloader(dependency, repoPath, replacements, line.hasOption("latest"),
+                    line.hasOption("pre"), line.hasOption("lenient"));
+            } else if (dependency.split(":").length == 3) {
+                artifactDownloader = new WheelsDownloader(dependency, repoPath, line.hasOption("lenient"));
+            } else {
+                String errMsg = "Unable to parse the dependency "
+                    + dependency
+                    + ".\nThe format of dependency should be either <module>:<revision> for source distribution "
+                    + "or <module>:<revision>:<classifier> for Wheels.";
+
+                if (line.hasOption("lenient")) {
```

Comment:
Does the lenient option matter here? I think the Exception should still be thrown even if the lenient option was specified.

Reply:
Adding lenient option will continue to import the following deps even one dep is failed to import,  we just write the error message to log and user will see that. If an Exception is thrown, then the importing will not continue.

Reply:
Ah gotcha. I forgot users could specify multiple dependencies when I was writing this comment.